cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/statements/ForEachStatement.kt

```diff
@@ -25,22 +25,63 @@
  */
 package de.fraunhofer.aisec.cpg.graph.statements
 
-import de.fraunhofer.aisec.cpg.graph.AST
+import de.fraunhofer.aisec.cpg.graph.*
+import de.fraunhofer.aisec.cpg.graph.edge.PropertyEdge
+import de.fraunhofer.aisec.cpg.graph.statements.expressions.DeclaredReferenceExpression
 import java.util.Objects
 
-class ForEachStatement : Statement() {
+class ForEachStatement : Statement(), BranchingNode, StatementHolder {
     /**
      * This field contains the iteration variable of the loop. It can be either a new variable
      * declaration or a reference to an existing variable.
      */
-    @AST var variable: Statement? = null
+    @AST
+    var variable: Statement? = null
+        set(value) {
+            if (value is DeclaredReferenceExpression) {
+                value.access = AccessValues.WRITE
+            }
+            field = value
+        }
 
     /** This field contains the iteration subject of the loop. */
     @AST var iterable: Statement? = null
 
     /** This field contains the body of the loop. */
     @AST var statement: Statement? = null
 
+    override val branchingDecision: Node?
+        get() = iterable
+
+    override var statementEdges: MutableList<PropertyEdge<Statement>>
+        get() {
+            val statements = mutableListOf<PropertyEdge<Statement>>()
+            variable?.let { statements.add(PropertyEdge(this, it)) }
+            iterable?.let { statements.add(PropertyEdge(this, it)) }
+            statement?.let { statements.add(PropertyEdge(this, it)) }
```

Comment:
this includes the compound statement itself (not the contents of it), why? this whole thing confuses me a little bit :)

Reply:
Maybe I made a mistake in Fluent-DSL definition. Somehow, this worked for me :D Feel free to fix the `forEachStmt` function in `Fluent.kt` and remove this code.

Reply:
I will have a look at it.